### PR TITLE
Added the what's new for iris 2.1

### PR DESCRIPTION
**Pull request by pelson**
Adds a what's new for a v2.1 release.
#3038 relates.

### CODE REVIEW

File: docs/iris/src/whatsnew/2.1.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+This document explains the new/changed features of Iris in version 2.1
+(:doc:`older "What's New" release notes can be found here<index>`.)
+
+
+Iris 2.1 Dependency updates
+===========================
+
+* The `cf_units <https://github.com/SciTools/cf_units>`_ dependency
+  was updated to cf_units ``v2.0``.
+  cf_units v2 is almost entirely backwards compatible with v1, however the
```
Comment (bjlittle) [inline docs/iris/src/whatsnew/2.1.rst]: `... backwards compatible with v1. However, ...`

File: docs/iris/src/whatsnew/2.1.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  calendar.
+* Iris updated its time-handling functionality from the
+  `netcdf4-python <http://unidata.github.io/netcdf4-python/>`_
+  ``netcdftime`` implementation to the standalone module
+  `cftime <https://github.com/Unidata/cftime>`_.
+  cftime is entirely compatible with netcdftime, but some issues may
+  occur where users are constructing their own datetime objects.
+  In this situation, simply replacing ``netcdftime.datetime`` with
+  ``cftime.datetime`` should be sufficient.
+* Iris now requires version 2 of matplotlib, and ``>=1.14`` of numpy.
```
Comment (bjlittle) [inline docs/iris/src/whatsnew/2.1.rst]: `numpy` to `NumPy` to align with line [69](https://github.com/SciTools/iris/pull/3044/files#diff-c3be53e80cb81dd25ed76b0dd9acbb46R69)... or vice versa.

File: docs/iris/src/whatsnew/2.1.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+Iris 2.1 Features
+=================
+
+* Added ``repr_html`` functionality to the :class:`~iris.cube.Cube` to provide
+  a rich html representation of cubes in Jupyter notebooks.
+
+  .. image:: images/notebook_repr.png
+
+* Updated :func:`iris.cube.Cube.name` to return a STASH code if the cube has
+  one and no other valid names are present. This is now consistent with the
```
Comment (bjlittle) [inline docs/iris/src/whatsnew/2.1.rst]: `... one if no other ...`

File: docs/iris/src/whatsnew/2.1.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+* The partial collapse of multi-dimensional auxiliary coordinates is now
+  supported. Collapsed bounds span the range of the collapsed dimension(s).
+* Added new function :func:`iris.cube.CubeList.realise_data` to compute
+  multiple lazy values in a single operation, avoiding repeated re-loading of
+  data or re-calculation of expressions.
+* The methods :meth:`iris.cube.Cube.convert_units` and
+  :meth:`iris.coords.Coord.convert_units` no longer forcibly realise the cube
+  data or coordinate points/bounds. The converted values are now lazy arrays
+  if the originals were.
+* Added :meth:`iris.analysis.trajectory.interpolate` that allows you
```
Comment (bjlittle) [inline docs/iris/src/whatsnew/2.1.rst]: `... that allows you to ...`

**Comment by bjlittle** [comment]
@pelson Great :+1: 

Suggested some minor tweaks, otherwise LGTM.

**Comment by pelson** [comment]
This is where https://help.github.com/articles/committing-changes-to-a-pull-request-branch-created-from-a-fork/ would be useful 😉

**Pull request merged**